File: IOT/app/src/main/java/com/example/velm/iot/activities/LoginActivity.java
package com.example.velm.iot.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.example.velm.iot.R;


public class LoginActivity extends AppCompatActivity implements View.OnClickListener{

    private Context context;
    private Button buttonLogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        context = this;
        initView();
    }


    private void initView(){
         buttonLogin = (Button)findViewById(R.id.buttonLogin);
        buttonLogin.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {

        switch (view.getId()){

            case R.id.buttonLogin:
                Intent mainActivityIntent = new Intent(context,DashBoardActivity.class);
                startActivity(mainActivityIntent);
                finish();
                break;
            default:
                break;
        }

    }
}

File: CleanArchitectureAndroid/app/src/main/java/com/example/cleanarchitectureandroid/ui/login/LoginActivity.kt
package com.example.cleanarchitectureandroid.ui.login

import android.content.Intent
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import com.example.cleanarchitectureandroid.R
import com.example.cleanarchitectureandroid.databinding.ActivityLoginBinding
import com.example.cleanarchitectureandroid.ui.home.HomeActivity
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

class LoginActivity : DaggerAppCompatActivity() {

    lateinit var loginBinding: ActivityLoginBinding

    @Inject
    lateinit var viewModelFactory: ViewModelProvider.Factory

    lateinit var loginViewModel: LoginViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        loginBinding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        loginBinding.lifecycleOwner = this
        loginViewModel = ViewModelProviders.of(this,viewModelFactory).get(LoginViewModel::class.java)

        loginBinding.loginViewModel = loginViewModel

        observeChanges()
    }

    private fun observeChanges() {
        loginViewModel.loginSuccess.observe(this, {
            it?.let {
                startActivity(Intent(this@LoginActivity, HomeActivity::class.java))
                finish()
            }
        })
    }
}